javascript_nodejs/6.using-cards/bot.js

```diff
@@ -0,0 +1,287 @@
+            }
+        );
+    }
+    
+    createHeroCard() {
+        return CardFactory.heroCard(
+            'BotFramework Hero Card',
+            CardFactory.images(['https://sec.ch9.ms/ch9/7ff5/e07cfef0-aa3b-40bb-9baa-7c9ef8ff7ff5/buildreactionbotframework_960.jpg']),
+            CardFactory.actions([
+                {
+                    type: 'openUrl',
+                    title: 'Get Started',
+                    value: 'https://docs.microsoft.com/en-us/azure/bot-service/'
+                }
+            ])
+        );
+    }
+    
+    createReceiptCard() {
+        return CardFactory.receiptCard({
+            title: "John Doe",
+            facts: [
+                {
+                    key: 'Order Number',
+                    value: '1234'
+                },
+                {
+                    key: 'Payment Method',
+                    value: 'VISA 5555-****'
+                }
+            ],
+            items: [
+                {
+                    title: 'Data Transfer',
+                    price: '$38.45',
+                    quantity: 368,
+                    image: { url: 'https://github.com/amido/azure-vector-icons/raw/master/renders/traffic-manager.png' }
+                },
+                {
+                    title: 'App Service',
+                    price: '$45.00',
+                    quantity: 720,
+                    image: { url: 'https://github.com/amido/azure-vector-icons/raw/master/renders/cloud-service.png' }
+                }
+            ],
+            tax: '$7.50',
+            total: '$90.95',
+            buttons: CardFactory.actions([
+                {
+                    type: 'openUrl',
+                    title: 'More Information',
+                    value: 'https://azure.microsoft.com/en-us/pricing/details/bot-service/'
+                }
+            ])
+        })
+    }
+
+    createSignInCard() {
+        return CardFactory.signinCard(
+            'BotFramework Sign-in Card',
+            'https://login.microsoftonline.com',
+            'Sign-in'
+        );
+    }
+    
+    createThumbnailCard() {
+        return CardFactory.thumbnailCard(
+            'BotFramework Thumbnail Card',
+            [{ url: 'https://sec.ch9.ms/ch9/7ff5/e07cfef0-aa3b-40bb-9baa-7c9ef8ff7ff5/buildreactionbotframework_960.jpg' }],
+            [{
+                type: 'openUrl',
+                title: 'Get Started',
+                value: 'https://docs.microsoft.com/en-us/azure/bot-service/'
+            }],
+            {
+                subtitle: 'Your bots — wherever your users are talking',
+                text: 'Build and connect intelligent bots to interact with your users naturally wherever they are, from text/sms to Skype, Slack, Office 365 mail and other popular services.'
+            }
+        )
+    }
+    
+    createVideoCard() {
+        return CardFactory.videoCard(
+            'Big Buck Bunny',
```

Comment:
> 'Big Buck Bunny', [](start = 11, length = 18)

Please pick a less weird video. Pick a ScottGu or Satya speech or similar.

Reply:
More than happy to... This content was lifted from our v3 rich cards sample...